code/game/gamemodes/events/space_weather.dm

```diff
@@ -0,0 +1,252 @@
+	tags = list(TAG_DESTRUCTIVE, TAG_NEGATIVE)
+
+/datum/event/harmonic_feedback
+	startWhen = 5
+	announceWhen = 1
+	endWhen = 350
+
+/datum/event/harmonic_feedback/setup()
+	endWhen = rand(300, 450)
+
+/datum/event/harmonic_feedback/announce()
+	command_announcement.Announce("The ship is currently passing through intense gravitational wavefronts. They will heavily disrupt hull shields for a short duration.", "Harmonic Feedback Surge Anomaly")
+
+/datum/event/harmonic_feedback/end()
+	command_announcement.Announce("The gravitational wavefronts have passed.", "Harmonic Feedback Surge Anomaly")
+
+/datum/event/harmonic_feedback/tick() //around two seconds
+	for(var/obj/machinery/power/shield_generator/G in GLOB.machines)
+		G.take_damage(10, SHIELD_DAMTYPE_EM)
+
+/datum/storyevent/micro_debris
+	id = "micro debris"
+	name = "micro debris field"
+	weight = 0.9
+	parallel = FALSE
+	event_type = /datum/event/dust
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_DESTRUCTIVE, TAG_NEGATIVE)
+
+/datum/event/micro_debris
+	startWhen	= 2
+	announceWhen = 1
+	endWhen		= 60
+	var/list/debris_types = list(
+		/obj/effect/meteor/dust/glass=40,\
+		/obj/effect/meteor/dust/rods=30,\
+		/obj/effect/meteor/dust/metal=20,\
+		/obj/effect/meteor/dust/ice=10
+	)
+
+/datum/event/micro_debris/announce()
+	command_announcement.Announce("The ship is now passing through a micro debris field.", "Micro Debris Field Alert")
+
+/datum/event/micro_debris/end()
+	command_announcement.Announce("The ship has now passed through the micro debris field.", "Micro Debris Field Notice")
+
+/datum/event/micro_debris/tick()
+	if(prob(60))
+		var/debris = pickweight(debris_types)
+		var/start_side = pick(cardinal)
+		var/target_zlevel = pick(GLOB.maps_data.station_levels)
+		for(var/i in 0 to rand(1,3))
+			spawn_debris(debris, start_side, target_zlevel)
+
+/datum/event/micro_debris/proc/spawn_debris(debris, start_side, zlevel)
+	var/turf/start_turf = spaceDebrisStartLoc(start_side, zlevel)
+	var/turf/destination = spaceDebrisFinishLoc(start_side, zlevel)
+	var/obj/effect/meteor/M = new debris(start_turf)
+	M.dest = destination
+	walk_towards(M, M.dest, 1)
+	return
+
+/datum/storyevent/graveyard
+	id = "graveyard"
+	name = "Space Graveryard"
+	weight = 0.6
+	parallel = FALSE
+	event_type = /datum/event/graveyard
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/graveyard
+	startWhen = 2
+	announceWhen = 1
+	endWhen = 350
+
+/datum/event/graveyard/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/graveyard/start()
+	for(var/mob/living/carbon/human/H in GLOB.human_mob_list)
+		H.sanity.sanity_damage_modifier = H.sanity.sanity_damage_modifier * 1.5
+		H.sanity.sanity_view_damage_modifier = H.sanity.sanity_view_damage_modifier * 1.5
```

Comment:
That works, but I would add a global modifier, instead of changing each individual mob's variable.
Could be either literally "global sanity modifier", or "weather sanity modifier".
New mobs will spawn, but they'll be unaffected by that event.
And player may increase their sanity damage modifier in some way, but then this weather resets it to `initial()`.